src/domain/usecases/UpdateHcvDataValuesUseCase.ts

```diff
@@ -0,0 +1,150 @@
+import _ from "lodash";
+import { Async } from "domain/entities/Async";
+import { Id } from "domain/entities/Base";
+import { Result } from "domain/entities/Result";
+import logger from "utils/log";
+import { ProgramEventsRepository } from "domain/repositories/ProgramEventsRepository";
+import { EventDataValue, ProgramEvent } from "domain/entities/ProgramEvent";
+import { HcvSettingsRepository } from "domain/repositories/HcvSettingsRepository";
+import { HcvSettings } from "domain/entities/HcvSettings";
+import { Maybe } from "utils/ts-utils";
+import { ProgramStageRepository } from "domain/repositories/ProgramStageRepository";
+import { ProgramStage, ProgramStageDataElement } from "domain/entities/ProgramStage";
+import { HcvExportRepository } from "domain/repositories/HcvExportRepository";
+
+export class UpdateHcvDataValuesUseCase {
+    constructor(
+        private programEventsRepository: ProgramEventsRepository,
+        private hcvSettingsRepository: HcvSettingsRepository,
+        private programStageRepository: ProgramStageRepository,
```

Comment:
Not sure if we could use the more general parent repo, ProgramRepository, but I don't see anything conceptually wrong with using a more specific ProgramStageRepository.